types.hh

```diff
@@ -122,6 +122,15 @@ bool lexicographical_compare(TypesIterator types, InputIt1 first1, InputIt1 last
 // than zero when the first value is respectively smaller, equal or greater
 // than the second value.
 template <typename TypesIterator, typename InputIt1, typename InputIt2, typename Compare>
+requires requires (TypesIterator types, InputIt1 i1, InputIt2 i2, Compare cmp) {
+    { *types };
```

Comment:
We also have requirements w.r.t. the type returned by this, and the other iterators.

Unfortunately the standard lib doesn't have any iterator concepts that I know of.

Reply:
In fact I found std::input_iterator et al. I'll use them if libstdc++ provides them and tighten the concept generally.

Reply:
> We also have requirements w.r.t. the type returned by this, and the other iterators.

Actually, the function have a requirement about the type of *types, just that it be compatible with cmp().

Reply:
Was able to use std::input_iterator with some hassle